fanctrl.py

```diff
@@ -1,146 +1,209 @@
+            while True:
+                client_socket, _ = server_socket.accept()
+                try:
+                    # Receive data from the client
+                    data = client_socket.recv(4096).decode()
+                    print("Received command:", data)
+
+                    args = parser.parse_args(shlex.split(data))
+                    if args.strategy or args._strategy:
+                        strategy = args.strategy
+                        if strategy is None:
+                            strategy = args._strategy
+                        try:
+                            if strategy == "defaultStrategy":
+                                self.clearOverwrittenStrategy()
+                            else:
+                                self.overwriteStrategy(strategy)
+                            client_socket.sendall(self.getCurrentStrategy().name.encode())
+                        except InvalidStrategyException:
+                            client_socket.sendall(("Error: unknown strategy: " + strategy).encode())
+                    if args.pause:
+                        self.pause()
+                        client_socket.sendall("Success".encode())
+                    if args.resume:
+                        self.resume()
+                        client_socket.sendall("Success".encode())
+                    if args.query:
+                        client_socket.sendall(self.getCurrentStrategy().name.encode())
+                    if args.list_strategies:
+                        client_socket.sendall('\n'.join(self.configuration.getStrategies()).encode())
+                    if args.reload:
+                        self.configuration.reload()
+                        if self.overwrittenStrategy is not None:
+                            self.overwriteStrategy(self.overwrittenStrategy.name)
+                        client_socket.sendall("Success".encode())
+                except:
+                    pass
+                finally:
+                    client_socket.shutdown(socket.SHUT_WR)
+                    client_socket.close()
+        finally:
+            server_socket.close()
 
     def setSpeed(self, speed):
         self.speed = speed
         bashCommand = f"ectool fanduty {speed}"
         subprocess.run(bashCommand, stdout=subprocess.PIPE, shell=True)
 
-    def adaptSpeed(self):
-        currentTemp = self.temps[self._tempIndex]
-        currentTemp = min(
-            currentTemp, self.getMovingAverageTemperature(self.movingAverageInterval)
-        )
-        minPoint = self.speedCurve[0]
-        maxPoint = self.speedCurve[-1]
-        for e in self.speedCurve:
+    def getActualTemperature(self):
+        bashCommand = "ectool temps all"
+        rawOut = subprocess.run(bashCommand, stdout=subprocess.PIPE, shell=True, text=True).stdout
+        rawTemps = re.findall(r'\(= (\d+) C\)', rawOut)
+        temps = sorted([x for x in [int(x) for x in rawTemps] if x > 0], reverse=True)
```

Comment:
`reverse` should not be `True`

Reply:
indeed, I will keep it reversed but use the index 0 instead of -1 to keep readability